STM32Cube/Tasks/flight_phase.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * flight_phase.c
+ *
+ *  Created on: May 17, 2024
+ *      Author: joedo
+ */
+
+#include "flight_phase.h"
+#include "millis.h"
+
+#define BOOST_TIME_MS 15000 //time from inj valve open to motor bunout
+#define RECOVERY_DELAY_MS 30000 //time from inj valve open to recovery deployment
+
+enum FLIGHT_PHASE flight_state = PHASE_PRELAUNCH;
+float injOpenTime;
+float deploymentTime;
+
+EventGroupHandle_t flightPhaseEventsHandle = NULL;
+#define INJ_OPEN (xEventGroupGetBits(flightPhaseEventsHandle) && INJ_OPEN_BIT)
+#define RECOVERY_DEPLOYMENT (xEventGroupGetBits(flightPhaseEventsHandle) && RECOVERY_DEPLOYMENT_BIT)
+
+
+void flightPhaseInit()
+{
+	flightPhaseEventsHandle = xEventGroupCreate();
+	xEventGroupClearBits(flightPhaseEventsHandle, 0xFFFF); //clear out the enitire group since API doesn't specify if values are initialized to 0
+}
+
+void flightPhaseTask(void * argument)
+{
+	while(1)
```

Comment:
this seems kind of not ideal to have forever while loop? (even with the osDelayUntil at the end.) Can this thing be a state machine instead? Or set timers for the phase times if theyre known?

I have 0 background context on this module, just raising possible discussion

Reply:
Agreed, it seems highly inefficent, if you black box it the only time an external task should be allowed to know this things flight state is by calling a method, then each time the method is called it can compute this (it is just a few if statments so no computational delay concerns). The only thing that does need to run in this loop is the detection for setting the boost phase, I am wondering if its worth sacrificing the better abstraction of the flight state we have setup here, and just merge this with another thread to save compute, or even better, have another thread trigger the interupt when the conditions for boost phase is met, then the rest can be on demand checking if the flight state is querried.